source/extensions/quic_listeners/quiche/platform/quic_cert_utils_impl.cc

```diff
@@ -0,0 +1,63 @@
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_cert_utils_impl.h"
+
+#include "openssl/bytestring.h"
+
+namespace quic {
+
+// static
+bool QuicCertUtilsImpl::ExtractSubjectNameFromDERCert(QuicStringPiece cert,
+                                                      QuicStringPiece* subject_out) {
```

Comment:
> could this code be added to `source/extensions/transport_sockets/tls/utility.cc`, where we have all TLS certificate utilities, instead of here? 

source/extensions/transport_sockets/tls/utility.cc already has similar function: getSubjectFromCertificate(). I didn't use that one for QuicCertsUtils  because its return type is a new string while what we need is a string_view of the original input.

Reply:
> I know this is based on the internal code, but I think that returning subject (and `absl::nullopt` or empty string in case of failure) might be more in line with Envoy coding style.
The signature of this function has to match https://quiche.googlesource.com/quiche/+/envoy-integration/quic/platform/api/quic_cert_utils.h#17

Reply:
Good point, thanks!